src/components/sources/DocumentModal.tsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalCloseButton,
+  Text,
+  Link,
+  UnorderedList,
+  ListItem,
+  Box,
+  VStack,
+} from "@chakra-ui/react";
+
+interface DocumentModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  document: Record<string, any> | null;
+  isLoading: boolean;
+  isError: boolean;
+  error?: string;
+}
+
+const formatValue = (value: any): string => {
+  if (typeof value === "string") {
+    return value;
+  } else if (typeof value === "number" || typeof value === "boolean") {
+    return value.toString();
+  } else if (value instanceof Date) {
+    return value.toISOString();
+  } else if (Array.isArray(value)) {
```

Comment:
I don't think these last two `if` conditions are necessary given their return type. Atleast, after testing locally, I didn't come across a scenario where this is needed

Reply:
You are right, they would be useful if we had nested types, like arrays inside of arrays. We will not have that any time soon, so I removed it.